projects/reservoir/train_ks.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import dacite
+import logging
+import numpy as np
+from sklearn.linear_model import Ridge
+from sklearn.utils import shuffle
+
+import yaml
+from fv3fit.reservoir import (
+    ReservoirComputingReadout,
+    Reservoir,
+    ReservoirComputingModel,
+)
+from fv3fit.reservoir.transform import InputNoise
+from fv3fit.reservoir.config import ReservoirTrainingConfig
+from ks import KSConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "ks_config", type=str, help=("Config file for KS equation parameters")
+    )
+    parser.add_argument("train_config", type=str, help=("Config file for training"))
+    return parser
+
+
+def transform_inputs_to_reservoir_states(X, reservoir, input_noise: InputNoise):
+    reservoir_states = []
+    X_noised = X + input_noise.generate()
+    for x in X_noised:
+        reservoir.increment_state(x)
+        reservoir_states.append(reservoir.state)
+    return np.array(reservoir_states)
+
+
+if __name__ == "__main__":
+    parser = _get_parser()
+    args = parser.parse_args()
+    with open(args.ks_config, "r") as f:
+        ks_config = dacite.from_dict(KSConfig, yaml.safe_load(f))
+    with open(args.train_config, "r") as f:
+        train_config_dict = yaml.safe_load(f)
+        train_config = ReservoirTrainingConfig.from_dict(train_config_dict)
+
+    training_ts = ks_config.generate(
+        n_steps=train_config.n_samples + train_config.n_burn, seed=train_config.seed
+    )
+    training_ts_burnin, training_ts_keep = (
+        training_ts[: train_config.n_burn],
+        training_ts[train_config.n_burn :],
+    )
+
+    reservoir = Reservoir(train_config.reservoir_hyperparameters)
+    input_noise = InputNoise(
+        size=ks_config.N, stddev=train_config.input_noise, seed=train_config.seed
+    )
+
+    reservoir.synchronize(training_ts_burnin)
```

Comment:
The Wikner / Arcomano 2020 papers seem to do this differently- since the Arcomano paper indicates they add the noise to the synchronization times too I'll do it the same way.

Reply:
Sounds good -- thanks!